### PR TITLE
Tk multitenant

### PR DESCRIPTION
**Pull request by ypoplavs**
## Pull request description 
This draft introduces an option to deploy Testkube in multiple namespace in the the same k8s cluster. It deploys all TK components in Release namespace, except for `testkube-operator` which is by default deployed to `testkube-system` namespace, unless specified otherwise. 

A few points:
-  If a user chooses another namespace for the `testkube-operator`, they need to provide it in `values.yaml` file (default is `testkube-system`)
- If a user installs TK in another namespace, they need to disable `testkube-operator`: `--set testkube-operator.enabled=false` via CLI or in the `values.yaml`.

**Important**!
Testkube Operator creates ClusterRoles which cannot be changed. The Operator is deployed once with the first chart installation of Testkube.
Thus if you uninstall the first release of TK, it will uninstall TK Operator as well so it needs to be running.

Example of installation.
Fresh TK install:
`helm install testkube testkube/testkube -n testkube --create-namespace --debug`
OR
`helm upgrade testkube testkube/testkube -n testkube --create-namespace --debug`

TK install in another namespace in the same cluster:
`helm install testkube testkube/testkube -n your-namespace --create-namespace --debug  --set testkube-operator.enabled=false`

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

If TK is already running in your cluster, it will re-create the Operator in another namespace and replace Testkube API's ClusterRoles with Roles

## Fixes

-

### CODE REVIEW

File: charts/testkube-api/templates/role.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
+{{ if .Values.rbac.createRoles }}
 apiVersion: {{ include "global.capabilities.rbac.apiVersion" . }}
-kind: ClusterRole
+kind: Role
 metadata:
-  name: executors-cluster-role-{{ .Release.Name }}
+  name: executors-role-{{ .Release.Name }}-{{ .Release.Namespace }}
```
Comment (dejanzele) [inline charts/testkube-api/templates/role.yaml:5]: do we need the `-{{ .Release.Namespace }}` suffix now that is is namespace-scoped?

File: charts/testkube-api/templates/rolebinding.yaml [hunk 2]
```diff
@@ -1,11 +1,13 @@
+{{ if .Values.rbac.createRoleBindings }}
+
 apiVersion: {{ include "global.capabilities.rbac.apiVersion" . }}
-kind: ClusterRoleBinding
+kind: RoleBinding
 metadata:
-  name: executors-crb-{{ .Release.Name }}
+  name: executors-crb-{{ .Release.Name }}-{{ .Release.Namespace }}
```
Comment (dejanzele) [inline charts/testkube-api/templates/rolebinding.yaml:6]: do we need the `-{{ .Release.Namespace }}` suffix now that is is namespace-scoped?

File: charts/testkube-api/templates/rolebinding.yaml [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 roleRef:
   apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
-  name: executors-cluster-role-{{ .Release.Name }}
+  kind: Role
+  name: executors-role-{{ .Release.Name }}-{{ .Release.Namespace }}
```
Comment (dejanzele) [inline charts/testkube-api/templates/rolebinding.yaml:10]: do we need the `-{{ .Release.Namespace }}` suffix now that is is namespace-scoped?

File: charts/testkube-api/templates/role.yaml (hunk 1 above)
Comment (ypoplavs) [reply charts/testkube-api/templates/role.yaml:5]: I will remove it once fix for namespaces-scoped role is released. Now it's been added for testing purposes

File: charts/testkube-api/templates/rolebinding.yaml (hunk 3 above)
Comment (ypoplavs) [reply charts/testkube-api/templates/rolebinding.yaml:10]: I will remove it once fix for namespaces-scoped role is released. Now it's been added for testing purposes

**Comment by tkonieczny** [comment]
Things checked, and working:
- Single-namespace installation (default), cluster-wide permissions - being able to watch any namespace (backwards compatibility)
- Multi-namespace installation (testkube-api.multinamespace.enabled=true)
- Triggers in the installation namespace
- Triggers in additionalNamespaces
- Upgrade of existing installation - after upgrade it's not possible to watch any namespace than the installation namespace

Note:
It's NOT possible to set additionalNamespaces to namespace where another testkube instance is already installed. But, that is the only limitation I found

**Comment by ypoplavs** [comment]
Option to add `additionalNamespaces` is available now..

**Review by vsukhin — APPROVED** [review]

**Pull request merged**